modules/promotion/src/Plugin/Commerce/PromotionCondition/OrderAmountComparison.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\commerce_promotion\Plugin\Commerce\PromotionCondition;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides a 'Order: Percentage off' condition.
+ *
+ * @PromotionCondition(
+ *   id = "commerce_promotion_order_amount_comparison",
+ *   label = @Translation("Order Amount Comparison"),
+ *   target_entity_type = "commerce_order",
+ * )
+ */
+class OrderAmountComparison extends PromotionConditionBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'amount' => NULL,
+      // @todo expose the operator in form.
+      'operator' => '>',
+    ] + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form += parent::buildConfigurationForm($form, $form_state);
+
+    $form['amount'] = [
+      '#type' => 'commerce_price',
+      '#title' => t('Amount'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
+    $order = $this->getTargetEntity();
+    /** @var \Drupal\commerce_price\Price $total_price */
+    $total_price = $order->getTotalPrice();
+    /** @var \Drupal\commerce_price\Price $comparison_price */
+    $comparison_price = $this->configuration['amount'];
```

Comment:
Need to extend FunctionalJavascript to make sure this works for retrieving value. Does in code.